src/bp/core/logger/file-persister.ts

```diff
@@ -0,0 +1,125 @@
+import { Logger, LoggerEntry } from 'botpress/sdk'
+import fs from 'fs'
+import { injectable } from 'inversify'
+
+import { BotpressConfig } from 'core/config/botpress.config'
+import _ from 'lodash'
+import moment from 'moment'
+import ms from 'ms'
+import path from 'path'
+
+@injectable()
+export class LoggerFilePersister {
+  private readonly INTERVAL = ms('2s')
+
+  private batch: string[] = []
+  private intervalRef
+  private currentPromise
+  private maxFileSize!: number
+  private outputFolder!: string
+  private logger!: Logger
+
+  constructor() {}
+
+  async initialize(botpressConfig: BotpressConfig, logger: Logger) {
+    const logsConfig = _.get(botpressConfig, 'logs.fileOutput', {})
+    if (!logsConfig.enabled) {
+      return
+    }
+
+    this.logger = logger
+    this.outputFolder = logsConfig.folder
+    this.maxFileSize = 1024 * logsConfig.maxFileSize
+
+    if (process.env.DEBUG_LOGGER) {
+      this.logger.debug(`Saving logs to ${this.outputFolder}, keeping file size under ${logsConfig.maxFileSize} kb`)
+    }
+
+    if (await this._isOutputWritable()) {
+      this.start()
+    }
+  }
+
+  appendLog(entry: LoggerEntry) {
+    const bot = entry.botId ? `(bot: ${entry.botId})` : ''
+    this.batch.push(`${entry.timestamp} ${entry.scope} (${entry.level}) ${bot} ${entry.message}${entry.metadata}\n`)
+  }
+
+  start() {
+    if (this.intervalRef) {
+      return
+    }
+    this.intervalRef = setInterval(this._runTask, this.INTERVAL)
+  }
+
+  stop() {
+    clearInterval(this.intervalRef)
+    this.intervalRef = undefined
+  }
+
+  private async _isOutputWritable() {
+    const fullPath = path.resolve(this.outputFolder)
+
+    try {
+      await fs.existsSync(fullPath)
+      await fs.accessSync(fullPath, fs.constants.W_OK)
```

Comment:
Why do you use `await` here if it's sync method? [Official documentation](https://nodejs.org/api/fs.html#fs_fs_accesssync_path_mode) make it without  `await`. It can be bit confuse) Or I missed something?

Reply:
reflex, too many times we forget await haha